utils.js

```diff
@@ -101,10 +108,17 @@ const postDefinition = (res, acronym, definitions, channel_id, username, token)
 }
 
 const newDefinition = (res, acronym, definition, channel_id, token, username, app_id) => {
-  saveAcronym(acronym, definition, app_id)
-  postToChannel(WAS_DEFINED_MESSAGE(acronym, username, definition), channel_id, token)
-  addOneTransaction('acronymsDefined')
-  res.json({ text: THANK_YOU_MESSAGE(acronym) })
+  // Checking whether the exact definition for this acronym already exists first
+  const definitionSnapshot = await getDefinition(acronym, definition, app_id).once('value')
+  const definitionFromDB = definitionSnapshot.val()
+  if (definitionFromDB.toLowerCase() === definition.toLowerCase()) {
```

Comment:
did not check for italic, bold, etc. - unless a library exists to check all of those I think it would be complicated to implement.